server/http/helloapp/embedded_wish.go

```diff
@@ -0,0 +1,196 @@
+package helloapp
+
+import (
+	"net/http"
+
+	"github.com/mattermost/mattermost-plugin-apps/server/apps"
+	"github.com/mattermost/mattermost-plugin-apps/server/utils/httputils"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+func (h *helloapp) handleCreateEmbedded(w http.ResponseWriter, req *http.Request, claims *apps.JWTClaims, data *apps.CallRequest) (int, error) {
+	err := h.asBot(func(mmclient *model.Client4, botUserID string) error {
+		post := &model.Post{
+			Message:   "Debug form",
+			ChannelId: data.Context.ChannelID,
+			UserId:    botUserID,
+		}
+		post.AddProp("appID", AppID)
+		post.AddProp("dialog", h.getDialogSmallSample())
+
+		_, resp := mmclient.CreatePost(post)
+		if resp.Error != nil {
+			return resp.Error
+		}
+		return nil
+	})
+
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	httputils.WriteJSONStatus(w, http.StatusOK, nil)
+	return http.StatusOK, nil
+}
+
+func (h *helloapp) handleSubmitEmbedded(w http.ResponseWriter, req *http.Request, claims *apps.JWTClaims, data *apps.CallRequest) (int, error) {
+	response := apps.CallResponse{
+		Type: apps.ResponseTypeOK,
+		Data: make(map[string]interface{}),
+	}
+	post := &model.Post{
+		Message: "Submitted",
+		Props:   model.StringInterface{},
+	}
+	response.Data["post"] = post
+	// response := apps.CallResponse{
+	// 	Type:  apps.ResponseTypeError,
+	// 	Data:  make(map[string]interface{}),
+	// 	Error: "Some error",
+	// }
+
+	// errors := map[string]string{}
+	// for key := range data.Values.Data {
+	// 	errors[key] = "Some other error"
+	// }
+	// response.Data["errors"] = errors
+
+	httputils.WriteJSON(w, response)
+	return http.StatusOK, nil
+}
+
+func (h *helloapp) getDialogSmallSample() model.OpenDialogRequest {
+	return model.OpenDialogRequest{
+		URL: h.AppURL(PathWishSubmitEmbedded),
+		Dialog: model.Dialog{
+			Title:   "Title for Small Dialog Test",
+			IconURL: "http://www.mattermost.org/wp-content/uploads/2016/04/icon.png",
+			Elements: []model.DialogElement{
+				{
+					DisplayName: "Display Name",
+					Name:        "realname",
+					Type:        "text",
+					Default:     "default text",
+					Placeholder: "placeholder",
+					HelpText:    "This a regular input in an interactive dialog triggered by a test integration.",
+				},
+			},
+		},
+	}
+}
+
+func (h *helloapp) getDialogFullSample() model.OpenDialogRequest {
```

Comment:
It looks like this function is not called. Is this here to toggle between the two dialogs by editing the source here?

Reply:
Exactly. It is in case you want to trigger a fuller example.